### PR TITLE
OCPBUGS-19822 Additional Network and MetalLB cannot use same network

### PR DESCRIPTION
**Pull request by kquinn1204**
[OCPBUGS-19822]: Additional Network and MetalLB cannot use same network

Version(s): 4.11, 4.12, 4.13, 4.14, 4.15 main

Issue:https://issues.redhat.com/browse/OCPBUGS-19822

Link to docs preview: https://66003--docspreview.netlify.app/openshift-enterprise/latest/networking/metallb/about-metallb#additional_network_and_metallb_limitation_about-metallb-and-metallb-operator

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by kquinn1204** [comment]
@himank98 Can you PTAL at this PR raised for [https://issues.redhat.com/browse/OCPBUGS-19822](https://issues.redhat.com/browse/OCPBUGS-19822)

**Comment by kalexand-rh** [comment]
I've opened the `enterprise-4.15` branch. Please review your PR and update the applicable versions to also cover 4.15 if you expect that the changes in this PR will be valid in that branch.

File: modules/nw-metallb-infra-considerations.adoc [hunk 1]
```diff
@@ -21,3 +21,15 @@ For example, the following infrastructures can benefit from adding the MetalLB O
...
+
+To resolve this issue, you have a couple of options:
+
+* **Use Different Subnets**: To avoid connection failures, consider placing the MetalLB IP in a different subnet from the one where the source pod resides. By doing this, traffic from the source pod will take the default gateway, which is typically OVN-K. This way, the traffic can reach its destination by using the OVN overlay network, and the connection should work as expected.
+
+* **Switch to MACVLAN Interface**: Alternatively, you can opt to use a MACVLAN interface instead of a VLAN interface for your additional network configuration. MACVLAN interfaces provide a more isolated network environment and can help prevent conflicts with MetalLB configurations. This change in interface type can help you avoid the connection issues that might arise when using the same VLAN for MetalLB and an additional network interface on the source pod.
+
+You can address this by ensuring the MetalLB IP is in another subnet so the traffic coming from a pod would take the default gateway, which is OVN-K and reach the destination via the OVN overlay. Alternatively use a MACVLAN interface instead of a VLAN interface.
```
Comment (asood-rh) [inline modules/nw-metallb-infra-considerations.adoc:35]: Isn't this statement same as first two line items?

Comment (kquinn1204) [reply modules/nw-metallb-infra-considerations.adoc:35]: Yes my bad thanks

File: modules/nw-metallb-infra-considerations.adoc [hunk 2]
```diff
@@ -21,3 +21,15 @@ For example, the following infrastructures can benefit from adding the MetalLB O
...
+
+If you use the same VLAN for both MetalLB and an additional network interface configured on a source pod, it can lead to a connection failure. A connection failure happens when both the MetalLB IP and the source pod are on the same node.
+
+To resolve this issue, you have a couple of options:
+
+* **Use Different Subnets**: To avoid connection failures, consider placing the MetalLB IP in a different subnet from the one where the source pod resides. By doing this, traffic from the source pod will take the default gateway, which is typically OVN-K. This way, the traffic can reach its destination by using the OVN overlay network, and the connection should work as expected.
+
+* **Switch to MACVLAN Interface**: Alternatively, you can opt to use a MACVLAN interface instead of a VLAN interface for your additional network configuration. MACVLAN interfaces provide a more isolated network environment and can help prevent conflicts with MetalLB configurations. This change in interface type can help you avoid the connection issues that might arise when using the same VLAN for MetalLB and an additional network interface on the source pod.
```
[the remark this answers is not in the document]
Comment (kquinn1204) [reply modules/nw-metallb-infra-considerations.adoc:33]: OK should I will remove the **Switch to MACVLAN Interface** section.

File: modules/nw-metallb-layer2-limitations.adoc [hunk 3]
```diff
@@ -34,3 +34,10 @@ The old node can continue to forward traffic for outdated clients until their ca
 
 During an unplanned failover, the service IPs are unreachable until the outdated clients refresh their cache entries.
 
+[id="additional_network_and_metallb_limitation_{context}"]
+== Additional Network and MetalLB cannot use same network
+
+Using the same VLAN for both MetalLB and an additional network interface set up on a source pod may result in a connection failure. This occurs when both the MetalLB IP and the source pod reside on the same node.
+
+To avoid connection failures, place the MetalLB IP in a different subnet from the one where the source pod resides. This configuration ensures that traffic from the source pod will take the default gateway, which is typically OVN-Kubernetes. Consequently, the traffic can effectively reach its destination by using the OVN overlay network, ensuring that the connection functions as intended.
+
```
Comment (JoeAldinger) [inline modules/nw-metallb-layer2-limitations.adoc:43]: Looks like an extra space here.

**Review by JoeAldinger — COMMENTED** [review]
Small suggestion to eliminate an extra space and wondering if this goes into 4.15 too?

File: modules/nw-metallb-infra-considerations.adoc [hunk 4]
```diff
@@ -24,12 +24,6 @@ MetalLB Operator and MetalLB are supported with the OpenShift SDN and OVN-Kubern
...
-
-* **Switch to MACVLAN Interface**: Alternatively, you can opt to use a MACVLAN interface instead of a VLAN interface for your additional network configuration. MACVLAN interfaces provide a more isolated network environment and can help prevent conflicts with MetalLB configurations. This change in interface type can help you avoid the connection issues that might arise when using the same VLAN for MetalLB and an additional network interface on the source pod.
-
-You can address this by ensuring the MetalLB IP is in another subnet so the traffic coming from a pod would take the default gateway, which is OVN-K and reach the destination via the OVN overlay. Alternatively use a MACVLAN interface instead of a VLAN interface.
\ No newline at end of file
+To avoid connection failures, place the MetalLB IP in a different subnet from the one where the source pod resides. This configuration ensures that traffic from the source pod will take the default gateway, which is typically OVN-Kubernetes. Consequently, the traffic can effectively reach its destination by using the OVN overlay network, ensuring that the connection functions as intended.
```
Comment (himank98) [inline modules/nw-metallb-infra-considerations.adoc]: This is not limited to OVN-K but also in SDN. The source pod will take the default gateway irrespective of CNI

**Comment by kquinn1204** [comment]
@JoeAldinger I made the changes as requested hopefully OK to merge now. Thanks

**Pull request merged**